thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/monitor/MonitorTaskRunner.java

```diff
@@ -120,11 +134,62 @@ private void executeMonitorUpdate(MonitorTaskInfo monitorTaskInfo) {
 
       // update detection health
       updateDetectionHealth();
+
+      // disable alerts that failed consecutively for a long time
+      disableLongFailedAlerts();
+
     } catch (Exception e) {
       LOG.error("Exception in monitor update task", e);
     }
   }
 
+  /**
+   * Disable the alert if it was updated before {@MAX_TASK_FAIL_DAYS} but there is no success run since then.
```

Comment:
What if the alert was updated(after receiving the below notification) and it still continues to fail? We don't want to disable & notify them again?

Reply:
Yes we want to handle it more conservatively. Give them 30 days grace period.